web/src/containers/activity/All.js

```diff
@@ -9,70 +8,59 @@ import {
-  <Waypoint id="activities-overview">
-    <Section isNumbered id="activities" resource="activities">
-      <Waypoint id="activities-list" />
-      <Subsection id="activities-list" resource="activities.list" open>
-        <NameAndFundingSourceReview
-          item={activities[0]}
-          index={-1}
-          disableExpand
-        />
-        <FormAndReviewList
-          addButtonText="Add another activity"
-          allowDeleteAll
-          list={activities.slice(1)}
-          className="ds-u-border-bottom--0"
-          collapsed={NameAndFundingSourceReview}
-          expanded={NameAndFundingSourceForm}
-          noDataMessage={false}
-          onAddClick={add}
-          handleChange={update}
-          onDeleteClick={remove}
-        />
-      </Subsection>
-      {activityKeys.map((key, idx) => (
-        <Waypoint id={key} key={key}>
-          <EntryDetails aKey={key} num={idx + 1} />
-        </Waypoint>
-      ))}
-    </Section>
-  </Waypoint>
-);
+const All = () => {
+  const dispatch = useDispatch();
 
-All.propTypes = {
-  activityKeys: PropTypes.array.isRequired,
-  activities: PropTypes.array.isRequired,
-  add: PropTypes.func.isRequired,
-  remove: PropTypes.func.isRequired,
-  update: PropTypes.func.isRequired
-};
+  const { activities } = useSelector(state => ({
+    activities: selectAllActivities(state)
+  }));
 
-export const mapStateToProps = state => ({
-  activityKeys: selectActivityKeys(state),
-  activities: selectAllActivities(state)
-});
+  const add = () => dispatch(addActivity());
 
-export const mapDispatchToProps = {
-  add: addActivity,
-  remove: removeActivity,
-  update: updateActivity
-};
+  const update = () => {};
+
+  const remove = key => {
+    activities.forEach(({ key: activityKey }, i) => {
+      if (activityKey === key) {
+        dispatch(removeActivity(i));
+      }
+    });
+  };
 
-export { All as AllRaw };
-export default connect(
-  mapStateToProps,
-  mapDispatchToProps
-)(All);
+  return (
+    <Waypoint id="activities-overview">
+      <Section isNumbered id="activities" resource="activities">
+        <Waypoint id="activities-list" />
+        <Subsection id="activities-list" resource="activities.list" open>
+          <NameAndFundingSourceReview
+            item={activities[0]}
+            index={-1}
+            disableExpand
+          />
+          <FormAndReviewList
+            addButtonText="Add another activity"
+            allowDeleteAll
+            list={activities.slice(1)}
+            className="ds-u-border-bottom--0"
+            collapsed={NameAndFundingSourceReview}
+            expanded={NameAndFundingSourceForm}
+            noDataMessage={false}
+            onAddClick={add}
+            handleChange={update}
+            onDeleteClick={remove}
+          />
+        </Subsection>
+        {activities.map((activity, index) => (
+          <Waypoint id={activity.key} key={activity.key}>
+            <EntryDetails activity={activity} index={index} />
```

Comment:
Instead of passing an activity key and number into `EntryDetails`, we now just pass the activity itself along with its index in the array. It's silly that we made each component down the line go fetch the activity again based on the key, but that's what we have/had.